app/src/main/kotlin/org/cru/godtools/ui/login/LoginLayout.kt

```diff
@@ -145,14 +153,26 @@ fun LoginLayout(createAccount: Boolean = false, onEvent: (event: LoginLayoutEven
 
 @Composable
 private fun LoginError(error: LoginResponse.Error?, onDismiss: () -> Unit) {
+    val context = LocalContext.current
+    val isConnected by remember { context.isConnectedFlow() }.collectAsState(true)
```

Comment:
this is business logic in the view layer, you should move detection of if they are offline to the `GodToolsAccountManager` login logic that produces the `LoginResponse.Error`.

What I had said about this on slack last week:

> we could either try and introduce an `AuthenticationException.DeviceOffline` exception we handle, and/or extend the else logic to be something like:
```
else -> when {
    isOffline -> LoginResponse.Error.DeviceOffline
    else -> LoginResponse.Error()
}
```